src/VirtualClient/VirtualClient.Common/SSHClient/ISshClientProxy.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Common
+{
+    using System;
+    using Renci.SshNet;
+
+    /// <summary>
+    /// Proxy for a Ssh Client.
+    /// </summary>
+    public interface ISshClientProxy : IDisposable
+    {
+        /// <summary>
+        /// Remote connection information of the SshClient.
+        /// </summary>
+        ConnectionInfo ConnectionInfo { get; }
+
+        /// <summary>
+        /// Creates the command to be executed.
+        /// </summary>
+        /// <param name="commandText">The command text.</param>
+        /// <returns>A Ssh Command object.</returns>
+        ISshCommandProxy CreateCommand(string commandText);
```

Comment:
Is it create command or run command?

Reply:
It just creates a command. We can run the commands independently.